### PR TITLE
Core: Make PropertyOwner remove all its properties before destruction…

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Make PropertyOwner remove all its properties before destruction to ensure that PropertyOwnerObserver::onWillRemoveProperty is called.

Otherwise there can be calls to onWillAddProperty but not to onWillRemoveProperty

### CODE REVIEW

**Comment by petersteneteg** [comment]
Where did you run into this? 
You might want to fix operator= also? that only does properties.clear()

**Comment by ResearchDaniel** [comment]
Property-synchronization with webbrowser. I need to remove widgets when a processor is removed. 
=operator does not use add either from what I could tell, not sure if it counts as add/remove. I’ll look into it a bit more

**Comment by petersteneteg** [comment]
https://github.com/inviwo/inviwo/blob/99ffbae57747d92f7a4471b62e9f555d9bf981e9/src/core/properties/propertyowner.cpp#L64
this must result in an add notification?

**Comment by petersteneteg** [comment]
https://github.com/inviwo/inviwo/blob/99ffbae57747d92f7a4471b62e9f555d9bf981e9/src/core/properties/propertyowner.cpp#L62 Humm I wounder if this is even correct? usually we call 
prop1 = rhs.prop1;
prop2 = rhs.prop2;
in the assignment operator of the derived class, but if we have done properties.clear() in this first those properties will not be listed anymore?

**Comment by ResearchDaniel** [comment]
Fixed

File: src/core/properties/propertyowner.cpp [hunk 1]
```diff
@@ -50,11 +50,20 @@ PropertyOwner::PropertyOwner(const PropertyOwner& rhs)
     for (const auto& p : rhs.ownedProperties_) addProperty(p->clone());
 }
 
+PropertyOwner::~PropertyOwner() {
+    while (begin() != end()) {
```
Comment (petersteneteg) [inline src/core/properties/propertyowner.cpp]: This crashes on osx...

Comment (ResearchDaniel) [reply src/core/properties/propertyowner.cpp]: Should be fixed now.

File: include/inviwo/core/processors/canvasprocessor.h [hunk 2]
```diff
@@ -92,6 +92,7 @@ class IVW_CORE_API CanvasProcessor : public Processor, public ProcessorWidgetMet
 public:
     ImageInport inport_;
 
+    CompositeProperty inputSize_;
```
Comment (petersteneteg) [inline include/inviwo/core/processors/canvasprocessor.h:95]: Will this not create a order warning in the constructor?

**Review by petersteneteg — APPROVED** [review]

File: include/inviwo/core/processors/canvasprocessor.h (hunk 2 above)
Comment (ResearchDaniel) [reply include/inviwo/core/processors/canvasprocessor.h:95]: Yes, fixed it

**Pull request merged**